### ISSUE TITLE
Cannot create Service type LoadBalancer in EKS clusters

### ISSUE THREAD

**Issue by dkoshkin** [kubernetes-sigs/cluster-api-provider-aws#3329]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
1. Create a cluster with https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v1.2.0/cluster-template-eks.yaml with name `my-cluster`
2. Create a `Deployment` and a `Service` with type `LoadBalancer`
```
apiVersion: v1
kind: Namespace
metadata:
  name: e2e-loadbalancer-created
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: example-deployment
  namespace: e2e-loadbalancer-created
  labels:
    app: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: docker.io/library/nginx:stable
        ports:
        - containerPort: 80
---
apiVersion: v1
kind: Service
metadata:
  name: example-service
  namespace: e2e-loadbalancer-created
spec:
  selector:
    app: nginx
  ports:
    - port: 8765
      targetPort: 80
  type: LoadBalancer
```
---

Saw an error in the `Service`
```
Error syncing load balancer: failed to ensure load balancer: could not find any suitable subnets for creating the ELB
```

And when enabled controller-manager logs:
```
No tagged subnets found; will fall-back to the current subnet only.
```

I saw these tags on the Subnets which at first looked correct as it had both `kubernetes.io/role/elb` and `kubernetes.io/cluster/<clusterName>`

```
kubernetes.io/role/elb	1
kubernetes.io/cluster/my-cluster shared
sigs.k8s.io/cluster-api-provider-aws/role	public
sigs.k8s.io/cluster-api-provider-aws/cluster/my-cluster	owned
Name my-cluster-subnet-public-us-west-2b
```

**What did you expect to happen:**
An ELB for the `Service` should be created and ` curl <ELB>:8765` should return an nginx  page.

---

The `No tagged subnets found` was interesting so I looked more into the Subnet tags.
Turns out the tag `kubernetes.io/cluster/my-cluster` is not correct.
The EKS clusters generated by CAPA are named like `<namespace>_<cluster_name>-control-plane` (in AWS console)

In my case as soon as I tagged a Subnet with `kubernetes.io/cluster/default_my-cluster-control-plane=shared` it created the ELB.
Once I did that I could create both public and private ELBs `Services`.

But those ELBs never became healthy because no instances were assigned to it.
The EC2 instances also need to be fixed to have a tag `kubernetes.io/cluster/default_my-cluster-control-plane=owned`

---
Surprisingly the SecurityGroup `...control-plane` that is also assigned to the EC2 instance did have the correct tag `kubernetes.io/cluster/default_my-cluster-control-plane=owned`.

I cannot just set the correct tag in `additionalTags` because the EC2 instances do not come up when it has both `kubernetes.io/cluster/default_my-cluster-control-plane=shared` and `kubernetes.io/cluster/my-cluster=shared`, the cloud controller in the Kubelet fails to initialize.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
Initially reported in https://kubernetes.slack.com/archives/CD6U2V71N/p1647565022221199

**TL;DR it looks like when tagging Subnets and EC2 instances we need to use `<namespace>_<cluster_name>-control-plane` instead of what just the user provided `<cluster_name>`.**

**Environment:**

- Cluster-api-provider-aws version: 
   `v1.2.0`
- Kubernetes version: (use `kubectl version`): 
   `v1.21.6`
- OS (e.g. from `/etc/os-release`):

**Comment by pydctw** [comment]
> Looking at the PR, I think we would need a similar change for EC2 instances and tag with KubernetesClusterName?

Any reason EC2 instances need to be tagged with `kubernetes.io/cluster/EKS_CLUSTER_NAME=owned` instead of `kubernetes.io/cluster/CAPI_CLUSTER_NAME=owned`?

I know it could be confusing that Cluster and EKSCluster have different names in EKS but wondering if you see any issue with the current tag.

cc @richardcase

**Comment by dkoshkin** [comment]
>Any reason EC2 instances need to be tagged with kubernetes.io/cluster/EKS_CLUSTER_NAME=owned instead of kubernetes.io/cluster/CAPI_CLUSTER_NAME=owned?

(From my testing) Yes the tags need to match in order for the cloud-controller to be able to find the EC2 instances, otherwise the created ELB did not have instances backing it.

**Comment by sedefsavas** [comment]
FYI, in CAPA-managed clusters, we add `kubernetes.io/cluster/CLUSTER_NAME=owned` tag to instances and service type of LB is working there.

After the fix, would be good to append LB creation to an existing e2e test. We have `verifyElbExists` test for CAPA-managed clusters, but that is missing the check if instance is attached to the LB or not, so would have missed this particular bug.

**Comment by LucaLanziani** [comment]
@faiq I can see that you are working in this, do you mind assign the issue to yourself :pray: . 
This will help when people (like me) are looking for new issues to work on.

**Comment by richardcase** [comment]
@faiq  - its also good to do this as well when you start to work on an issue:

/lifecycle active

**Comment by richardcase** [comment]
@faiq - just checking in to see how you are getting on with this?

**Comment by faiq** [comment]
This was solved with this PR https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3343. Unclear why the automation didn't close it
